runtime/connectors/gcs/gcs.go

```diff
@@ -100,10 +98,11 @@ func (c connector) ConsumeAsFiles(ctx context.Context, env *connectors.Env, sour
 	return rillblob.FetchFileNames(ctx, bucketObj, fetchConfigs, glob, bucket)
 }
 
-func gcsURLParts(path string) (string, string, string, error) {
-	u, err := url.Parse(path)
-	if err != nil {
-		return "", "", "", err
+func gcsURLParts(path string) (string, string, error) {
+	trimmedPath := strings.Replace(path, "gs://", "", 1)
+	bucket, glob, found := strings.Cut(trimmedPath, "/")
+	if !found {
+		return "", "", fmt.Errorf("failed to parse path %s", path)
 	}
-	return u.Host, strings.Replace(u.Path, "/", "", 1), fileutil.FullExt(u.Path), nil
+	return bucket, glob, nil
 }
```

Comment:
At this point, we don't know if `path` is well-formed, so if not using `url.Parse`, it needs to be more defensively parsed.

For example, if the input is `whoops/bags://d/match.csv`, it will actually fetch `bad/match.csv` from the `whoops` bucket!

Also, let's add a comment describing why we're not using `url.Parse`

Reply:
Same applies for `s3`. Maybe we should add a `globurl` util package in `runtime/pkg` with a function like `url.Parse` that works on URLs containing globs?

Reply:
Makes sense. Will change.